File: Stm4/utils/TrieMetadata.java
package utils;


public class TrieMetadata {
	byte prefix_length ;
	long prefix ;
	
	public TrieMetadata (){}
	
	public TrieMetadata (byte prefix_length, long prefix){
		this.prefix = prefix; 
		this.prefix_length = prefix_length;
	}
}

File: GrainedC++/grained C++/TrieMetadata.cpp
#include "TrieMetadata.h"
	
TrieMetadata::TrieMetadata (){};

	TrieMetadata::TrieMetadata (char prefix_length, long prefix){
		this->prefix = prefix; 
		this->prefix_length = prefix_length;
	}

TrieMetadata::~TrieMetadata(void)
{
}
